src/jsonrpc/info.rs

```diff
@@ -369,32 +370,8 @@ impl GetNodeVersionResult {
             version: String::new(),
             database_version: String::new(),
             git_commit: String::new(),
-            vm_versions: VmVersions::default(),
-        }
-    }
-}
-
-/// ref. <https://docs.avax.network/build/avalanchego-apis/info/#infogetnodeversion>
-#[derive(Debug, Serialize, Deserialize, Eq, PartialEq, Clone)]
-#[serde(rename_all = "camelCase")]
-pub struct VmVersions {
-    pub avm: String,
-    pub evm: String,
-    pub platform: String,
-}
```

Comment:
🤔  I think we would beneficial to do something like this:
```rust
/// ref. <https://docs.avax.network/build/avalanchego-apis/info/#infogetnodeversion>
#[derive(Debug, Serialize, Deserialize, Eq, PartialEq, Clone)]
#[serde(rename_all = "camelCase")]
pub struct VmVersions {
    pub avm: String,
    pub evm: String,
    pub platform: String,
    #[serde(flatten)]
    pub subnet_map: HashMap<String, String>,
}
```
Each key in a `HashMap` is an allocation. If we know that the `avm`, `evm`, and `platform` are always going to be present, then we shouldn't allocate their names. This also simplifies things in terms of accessing `avm`, `evm`, and `platform` and has the added benefit of making the adoption of this breaking change much simpler. 

There are more ways to make this more performant and type-safe (with the above suggestion, you could technically have `String::from("avm")` as a key in the `subnet_map` that would overwrite `thing.avm` during serialization), but I think this should be fine for now.

Reply:
Elegant solution :+1: 

I named the HashMap `subnets`